src/frontend/README.md

```diff
@@ -1,7 +1,128 @@
+
+### Initialization
+
+The OpenTelemetry SDK is initialized in `main` using the `InitTraceProvider` function.
+
+```go
+func InitTracerProvider() *sdktrace.TracerProvider {
+    ctx := context.Background()
+
+    exporter, err := otlptracegrpc.New(ctx)
+    if err != nil {
+        log.Fatal(err)
+    }
+    tp := sdktrace.NewTracerProvider(
+        sdktrace.WithSampler(sdktrace.AlwaysSample()),
+        sdktrace.WithBatcher(exporter),
+    )
+    otel.SetTracerProvider(tp)
+    otel.SetTextMapPropagator(propagation.NewCompositeTextMapPropagator(propagation.TraceContext{}, propagation.Baggage{}))
+    return tp
+}
+```
+
+Services should call `TraceProvider.shutdown()` when the service is shutdown to
+ensure all spans are exported.
+This service makes that call as part of a deferred function in `main`.
+
+```go
+    // Initialize OpenTelemetry Tracing
+    tp := InitTracerProvider()
+    defer func() {
+        if err := tp.Shutdown(context.Background()); err != nil {
+            log.Printf("Error shutting down tracer provider: %v", err)
+        }
+    }()
+```
+
+### HTTP instrumentation
+
+This service receives HTTP requests, controlled by the gorilla/mux Router.
+The following routes are defined by the frontend:
+
+| Path              | Method | Use                               |
+|-------------------|--------|-----------------------------------|
+| `/`               | GET    | Main index page                   |
+| `/cart`           | GET    | View Cart                         |
+| `/cart`           | POST   | Add to Cart                       |
+| `/cart/checktout` | POST   | Place Order                       |
+| `/cart/empty`     | POST   | Empty Cart                        |
+| `/logout`         | GET    | Logout                            |
+| `/product/{id}`   | GET    | View Product                      |
+| `/setCurrency`    | POST   | Set Currency                      |
+| `/static/`        | *      | Static resources                  |
+| `/robots.txt`     | *      | Search engine response (disallow) |
+| `/_healthz`       | *      | Health check (ok)                 |
+
+These requests are instrumented in the main function as part of the router's definition.
+
+```go
+    // Add OpenTelemetry instrumentation to incoming HTTP requests controlled by the gorilla/mux Router.
+    r.Use(otelmux.Middleware("server"))
+```
+
+### gRPC instrumentation
+
+This service will issue several outgoing gRPC calls, which have instrumentation
+hooks added in the `mustConnGRPC` function.
+
+```go
+    // add OpenTelemetry instrumentation to outgoing gRPC requests
+    var err error
+    *conn, err = grpc.DialContext(ctx, addr,
+        grpc.WithTransportCredentials(insecure.NewCredentials()),
+        grpc.WithUnaryInterceptor(otelgrpc.UnaryClientInterceptor(otelgrpc.WithTracerProvider(otel.GetTracerProvider()))),
+        grpc.WithStreamInterceptor(otelgrpc.StreamClientInterceptor(otelgrpc.WithTracerProvider(otel.GetTracerProvider()))),
```

Comment:
These lines are different than what's currently in the `main` branch. I assume there's another PR bringing this to an equilibrium